src/test/java/bio/terra/service/iam/sam/SamIamTest.java

```diff
@@ -1,16 +1,94 @@
 package bio.terra.service.iam.sam;
 
 
+import bio.terra.app.configuration.SamConfiguration;
 import bio.terra.common.category.Unit;
+import bio.terra.model.PolicyModel;
+import bio.terra.model.RepositoryStatusModelSystems;
+import bio.terra.model.UserStatusInfo;
+import bio.terra.service.configuration.ConfigEnum;
+import bio.terra.service.configuration.ConfigurationService;
+import bio.terra.service.iam.AuthenticatedUserRequest;
+import bio.terra.service.iam.IamAction;
+import bio.terra.service.iam.IamResourceType;
+import bio.terra.service.iam.IamRole;
+import org.assertj.core.util.Maps;
+import org.broadinstitute.dsde.workbench.client.sam.ApiClient;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.broadinstitute.dsde.workbench.client.sam.api.GoogleApi;
+import org.broadinstitute.dsde.workbench.client.sam.api.ResourcesApi;
+import org.broadinstitute.dsde.workbench.client.sam.api.StatusApi;
+import org.broadinstitute.dsde.workbench.client.sam.api.UsersApi;
+import org.broadinstitute.dsde.workbench.client.sam.model.AccessPolicyMembership;
+import org.broadinstitute.dsde.workbench.client.sam.model.AccessPolicyResponseEntry;
 import org.broadinstitute.dsde.workbench.client.sam.model.ErrorReport;
+import org.broadinstitute.dsde.workbench.client.sam.model.ResourceAndAccessPolicy;
+import org.broadinstitute.dsde.workbench.client.sam.model.SystemStatus;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 @Category(Unit.class)
 public class SamIamTest {
 
+    @Mock
+    private SamConfiguration samConfig;
+    @Mock
+    private ConfigurationService configurationService;
+    @Mock
+    private ApiClient apiClient;
+    @Mock
+    private ResourcesApi samResourceApi = mock(ResourcesApi.class);
+    @Mock
+    private StatusApi samStatusApi = mock(StatusApi.class);
+    @Mock
+    private GoogleApi samGoogleApi = mock(GoogleApi.class);
+    @Mock
+    private UsersApi samUsersApi = mock(UsersApi.class);
+
+    @Mock
+    private AuthenticatedUserRequest userReq;
+
+    private SamIam samIam;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        samIam = spy(new SamIam(samConfig, configurationService));
```

Comment:
Using `spy` on the class under test can be tricky, and can lead to cases where your test is testing the mock. I think fixing it here could be done by splitting out the four `Api` factory methods from `SamIam` into a different class which would be injectable. The current pattern, of using `new XXX()` inside non-interface methods is not great anyway.

Reply:
Yeah, I wasn't sure.  I guess we could, I just felt like this might scope down the PR a little bit. I see your point about being careful with `spy`, that being said, I don't hate having an example of it here